js/coinsbit.js

```diff
@@ -0,0 +1,682 @@
+            //            open: '47030.33',
+            //            high: '47549.99',
+            //            low: '45571.08',
+            //            last: '46459.99',
+            //            volume: '8372.29022355',
+            //            deal: '388090406.73607745',
+            //          },
+            //          code: "200"
+            //     }
+            ticker = this.safeValue (response, 'result', {});
+        } else if (exchangeType === 'futures') {
+            // data = await this.futuresPublicGetTicker (request);
+        }
+        return this.parseTicker (ticker, market);
+    }
+
+    async fetchTickers (symbols = undefined, params = {}) {
+        const [exchangeType, request] = this.checkExchangeType ('fetchTickers', params);
+        await this.loadMarkets ();
+        let tickers = {};
+        if (exchangeType === 'spot') {
+            const response = await this.spotPublicGetTickers (request);
+            //     {
+            //          success: true,
+            //          message: "",
+            //          result: {
+            //            BTC_USDT: {
+            //               at: '1640028811',
+            //               ticker: {
+            //                 bid: '46448.49682539',
+            //                 ask: '46458.41093569',
+            //                 open: '46687.39',
+            //                 high: '47550',
+            //                 low: '45584.27',
+            //                 last: '46462.77',
+            //                 vol: '32.56275846',
+            //                 deal: '1510574.62551143',
+            //                 change: '-0'
+            //               }
+            //            },
+            //            ...
+            //          },
+            //          code: "200"
+            //     }
+            tickers = this.safeValue (response, 'result', {});
+        } else if (exchangeType === 'futures') {
+            // tickers = ..
+        }
+        return this.parseTickers (tickers, undefined, params);
+    }
+
+    // This modified parseTickers method needs to exist in base, to keep the KYES & pass them through parseTickers
+    parseTickers (tickers, symbols = undefined, params = {}) {
+        const result = [];
+        const values = Object.values (tickers || []);
+        const keys = Object.keys (tickers);
+        for (let i = 0; i < values.length; i++) {
+            result.push (this.extend (this.parseTicker (values[i], undefined, keys[i]), params));
+        }
+        return this.filterByArray (result, 'symbol', symbols);
+    }
+
+    parseTicker (ticker, market = undefined, key = undefined) {
+        let tickerFinal = {};
+        let timestamp = null;
+        if (market === undefined) {
+            // if arrived from parseTickers
+            market = this.market (key);
+            tickerFinal = ticker['ticker'];
+            timestamp = Precise.stringticker['at'] + '000';
```

Comment:
The timestamp is always an integer in milliseconds, not a string.
We should always sanitize the input with safe-methods and never access the keys like `ticker['at']` directly:
- https://github.com/ccxt/ccxt/blob/master/CONTRIBUTING.md#sanitizing-input-with-safe-methods

```JavaScript
const timestamp = this.safeTimestamp (ticker, 'at'); // will convert from seconds to milliseconds safely
```

Reply:
uh, that line was a complete mistake, forgot to fix.